### PR TITLE
Update dev mode to Sandbox mode throughout the plugin

### LINKED ISSUE #7974: Make "Sandbox mode" term used everywhere where "Dev mode" is used
### Description
Following up on a decision driven here paJDYF-bGl-p2 we want to use "sandbox mode" instead of "dev mode" in order to avoid confusion for our users

This is an additional issue to https://github.com/Automattic/woocommerce-payments/issues/7731 which is only taking care of the "selection mode", when this issue takes care of settings, banners and everywhere else to have the "dev mode" renamed to "sandbox mode" everywhere (including the dev tool).

### PR DESCRIPTION
**Pull request by dmallory42**
Fixes #7974

#### Changes proposed in this Pull Request

This PR updates all **front-facing** references in the plugin to "dev mode" to say "sandbox mode" which is aimed at making it more clear what the intention of this mode is, and to emphasise that live transactions can never occur in this mode. The full discussion is contained in the P2 thread and the latest designs: paJDYF-bGl-p2#comment-22402
 
I opted against updating code/comments in general because the concept of "dev mode" is quite firmly embedded into the plugin code and I think it's well understood by devs - the main purpose of this change is that the term sandbox mode is more easily understandable to front-end users of the plugin.

#### Testing instructions

* Onboard an account in sandbox mode (select this option when onboarding to be taken to the Stripe KYC flow).
* After onboarding, check each of the admin pages in WooPayments - anywhere where we'd previously have referred to dev mode should now instead reference sandbox mode.

TODO Post Merge

Docs will need to be updated prior to the next release. (Will chase this after PR is approved). https://woo.com/document/woopayments/testing-and-troubleshooting/dev-mode/

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8004` or branch name `update/dev-mode-labels` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=update/dev-mode-labels) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 7aa705f824514491ba468596f92083f3ed686a0a
- Build time: 2024-01-20 20:29:46 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Review by vladolaru — APPROVED** [review]
Thank you for working on this, Dan! The changes make sense and the testing instructions checked out!

There were a couple of more places that needed to be updated and a Jest snapshop. Took care of those myself.

File: client/overview/strings.tsx [hunk 1]
```diff
@@ -26,7 +26,7 @@ export default {
 				mixedString: sprintf(
 					/* translators: %1$s: WooPayments */
 					__(
-						'{{bold}}%1s is in dev mode.{{bold /}}. You need to set up a live %1s account before you can accept real transactions.',
+						'{{bold}}%1s is in sandbox mode.{{bold /}}. You need to set up a live %1s account before you can accept real transactions.',
```
Comment (emilyaudela) [inline client/overview/strings.tsx:29]: It looks like there is an extra period after `{{bold /}}`.

Comment (vladolaru) [reply client/overview/strings.tsx:29]: Oh! Good catch, Emily!

Comment (vladolaru) [reply client/overview/strings.tsx:29]: @dmallory42 Could we do a tiny subsequent PR to fix what Emily shared?

**Pull request merged**